File: languages/js/src/entity/anchor.ts
import * as proto from "../bridge/proto/anchor";

export class Anchor {
  id: number;
  blockRoots: string[];
  networks: AnchorNetwork[];
  root: string;
  status: string;

  constructor(
    id: number,
    blockRoots: string[],
    networks: AnchorNetwork[],
    root: string,
    status: string
  ) {
    this.id = id;
    this.blockRoots = blockRoots;
    this.networks = networks;
    this.root = root;
    this.status = status;
  }

  static fromProto(a: proto.Anchor): Anchor {
    return new Anchor(
      a.id, a.blockRoots,
      a.networks.map((n) => AnchorNetwork.fromProto(n)),
      a.root, a.status
    );
  }
}

export class AnchorNetwork {
  name: string;
  state: string;
  txHash: string;

  constructor(name: string, state: string, txHash: string) {
    this.name = name;
    this.state = state;
    this.txHash = txHash;
  }

  static fromProto(a: proto.AnchorNetwork): AnchorNetwork {
    return new AnchorNetwork(a.name, a.state, a.txHash);
  }

  toProto(): proto.AnchorNetwork {
    return proto.AnchorNetwork.fromPartial({ name: this.name, state: this.state, txHash: this.txHash });
  }
}

File: languages/python/bloock/client/entity/anchor.py
from __future__ import annotations
from typing import List
import bloock._bridge.proto.anchor_pb2 as proto


class Anchor:
    def __init__(
        self,
        id: int,
        block_roots: List[str],
        networks: List[AnchorNetwork],
        root: str,
        status: str,
    ) -> None:
        self.id = id
        self.block_roots = block_roots
        self.networks = networks
        self.root = root
        self.status = status

    @staticmethod
    def from_proto(anchor: proto.Anchor) -> Anchor:
        return Anchor(
            id=anchor.id,
            block_roots=list(anchor.block_roots),
            networks=list(map(lambda x: AnchorNetwork.from_proto(x), anchor.networks)),
            root=anchor.root,
            status=anchor.status,
        )

    def to_proto(self) -> proto.Anchor:
        return proto.Anchor(
            id=self.id,
            block_roots=list(self.block_roots),
            networks=list(map(lambda x: x.to_proto(), self.networks)),
            root=self.root,
            status=self.status,
        )


class AnchorNetwork:
    def __init__(self, name: str, state: str, tx_hash: str) -> None:
        self.name = name
        self.state = state
        self.tx_hash = tx_hash

    @staticmethod
    def from_proto(network: proto.AnchorNetwork) -> AnchorNetwork:
        return AnchorNetwork(
            name=network.name, state=network.state, tx_hash=network.tx_hash
        )

    def to_proto(self) -> proto.AnchorNetwork:
        return proto.AnchorNetwork(
            name=self.name, state=self.state, tx_hash=self.tx_hash
        )
